syntax = "proto3";
package gameServer;
service GameService{
	//Move will be used to move a character around a 2d map
	rpc Move(Direction) returns (Position) {}; 
	

	//attack will be a physical or magic attack and requires stamina or magic
	rpc MagicAttack(Player) returns (Player) {};

	//physical attack will be an attack with a weapon
	rpc PhysicalAttack(Player) returns (Player){};

	
	//Heal will heal a character and has to be done with magic
	rpc Heal(Player) returns (Player) {};

	//rpc DamageCalculation(Player) returns (Attack){};
}


message Empty {}


//the player object will have actions affect his/her stats.
message Player{
	map<string,int32> spellbook = 1;
	enum Stats{
		HEALTH = 0;
		STAMINA = 1;
		MANA = 2;
		LEVEL = 3;
		EXP = 4;
		XVAL = 5;
		YVAL = 6;
		NAME = 7;
		ATTACK = 8;
		DEFENSE = 9;
	}
	Stats stats = 2;
	
}


//direction input for a  player 
message Direction{
	string x_direction = 1;
	string y_direction = 2;
}
message Position{
	int32 x_position = 1;
	int32 y_position = 2;
}


//health status
message Health{
	int32 healthBar = 1;

}


//how much magic a player has
message Magic{
	int32 mana = 1;

}


//how much stamina a player has
message Stamina{
	int32 stamina = 1;
}

message Attack{
	int32 damage = 1;
	
}

message SpellBook{
	map<string,int32> spellbook = 1;

}


//pair to make a dictionary
message Pair{
	string key = 1;
	string value = 2;
}


//a dictionary of items or the players bag on their adventure
message Bag{
	repeated Pair equipment = 1;
}


